cli/base_commands.py

```diff
@@ -118,3 +124,44 @@ def get_help_string(self) -> str:
     async def execute_default(self, args, context):
         print('Invalid usage. Provide a subcommand.')
         return CommandResultNone()
+
+
+class ScanCommand(Command):
+    def get_help_string(self) -> str:
+        return 'Perform scan for TCAT devices'
+
+    async def execute_default(self, args, context):
+        if not (context['ble_sstream'] is None):
+            del context['ble_sstream']
+
+        tcat_devices = await ble_scanner.scan_tcat_devices()
+        if tcat_devices:
+            print('Found devices:\n')
+            for i, device in enumerate(tcat_devices):
+                print(f'{i + 1}: {device.name} - {device.address}')
+        else:
+            print('\nNo devices found.')
+            return CommandResultNone()
+
+        print('\nSelect the target number to connect to it.')
+        selected = get_int_in_range(1, len(tcat_devices))
+        device = tcat_devices[selected - 1]
+        print('Selected ', device)
+
+        ble_sstream = None
+
+        print(f'Connecting to {device}')
+        ble_stream = await BleStream.create(
+            device.address, BBTC_SERVICE_UUID, BBTC_TX_CHAR_UUID, BBTC_RX_CHAR_UUID
+        )
+        ble_sstream = BleStreamSecure(ble_stream)
+        ble_sstream.load_cert(
+            certfile=path.join('auth', 'commissioner_cert.pem'),
+            keyfile=path.join('auth', 'commissioner_key.pem'),
+            cafile=path.join('auth', 'ca_cert.pem'),
+        )
+
+        print('Setting up secure channel...')
+        await ble_sstream.do_handshake(hostname=SERVER_COMMON_NAME)
+        print('Done')
```

Comment:
should this be moved to external function to prevent doubling of the code?
https://github.com/pjasinski990/bbtc/blob/main/bbtc.py#L57